### PR TITLE
Move _load_subscriptions from profile to core

### PR DESCRIPTION
**Pull request by troydai**
Fix https://github.com/Azure/azure-cli/issues/6813

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yugangw-msft** [comment]
By looking at #6812, then this PR looks good to me, assuming it will address that as well, but please let me know otherwise.

File: src/azure-cli-core/azure/cli/core/_profile.py [hunk 1]
```diff
@@ -67,6 +67,14 @@
 _USER_ASSIGNED_IDENTITY = 'userAssignedIdentity'
 
 
+def _load_subscriptions(cli_ctx, all_clouds=False, refresh=False):
```
Comment (yugangw-msft) [inline src/azure-cli-core/azure/cli/core/_profile.py]: Suggest rename to `load_subscriptions`

**Comment by troydai** [comment]
Honestly, it is a private method. It shouldn't be dependable. I think we're moving towards the model to explicitly define the API we intend to expose.

File: src/azure-cli-core/azure/cli/core/_profile.py (hunk 1 above)
Comment (tjprescott) [reply src/azure-cli-core/azure/cli/core/_profile.py]: I agree, and if we do make it public, should it be under the "API" package as well so external teams can formally depend on it?

Comment (troydai) [reply src/azure-cli-core/azure/cli/core/_profile.py]: You have a point.

**Comment by troydai** [comment]
@tjprescott @yugangw-msft updated. another look?

**Pull request merged**